joomla/joomla-cms
# [6.0] Allow adding additional layout base path to form fields

### Summary of Changes
Currently the usage of form field layouts is very encapsulated in the FormField class, which means that it's not possible to change/add additional layout include paths via e.g. plugins.

There are two ways to add an alternative path
1. via XML attribute "layoutIncludePath"
2. via the ```$option``` attribute in the ```->renderField(...)``` or ```->renderFieldset(...)``` methods

### Testing Instructions
There are several ways to test this PR, one would be to create a plugin and manipulate the form with the onContentPrepareForm event. 

But to the test the raw functionality you can do the following:

- Create a folder "otherlayouts" in your Joomla! root
- Add the following folder structure ```joomla/form``` to the "otherlayouts" folder
- Copy the file ```layouts/joomla/form/renderfield.php``` to the new path
- Change this file somehow to see the difference when it's loaded (e.g. add some output like "Hello World" somewhere in the file)
- We will test now at the frontend registration, so open the form

#### Option 1:
- Open ```components/com_users/forms/registration.xml```
- Add after line 17 the following attribute: ```layoutIncludePath="otherlayouts"``` (it should be in the "name" field)
- Refresh registration, now you should see the "Hello World" from above at the "name" field

#### Option 2:
- Open ```components/com_users/tmpl/registration/default.php```
- Change line 44 to: ```<?php echo $this->form->renderFieldset($fieldset->name, ['layoutIncludePath' => JPATH_ROOT . '/otherlayouts/']); ?>```
- Refresh registration, now you should see the "Hello World" from above at each field
```
- Refresh registration, now you should see the "Hello World" from above at a "name" field

### Actual result BEFORE applying this Pull Request
- No alternative layout path possible


### Expected result AFTER applying this Pull Request
- Alternative layout path possible
